core/app/[locale]/(default)/account/addresses/_actions/create-address.ts

```diff
@@ -0,0 +1,263 @@
+import { TAGS } from '~/client/tags';
+import { FieldNameToFieldId } from '~/data-transformers/form-field-transformer/utils';
+
+import { type State } from './address-action';
+
+const AddCustomerAddressMutation = graphql(`
+  mutation AddCustomerAddressMutation($input: AddCustomerAddressInput!) {
+    customer {
+      addCustomerAddress(input: $input) {
+        errors {
+          ... on CustomerAddressCreationError {
+            message
+          }
+          ... on CustomerNotLoggedInError {
+            message
+          }
+          ... on ValidationError {
+            message
+            path
+          }
+        }
+        address {
+          entityId
+        }
+      }
+    }
+  }
+`);
+
+const stringToNumber = z.coerce.string().pipe(z.coerce.number());
+
+const inputSchema = z.object({
+  firstName: z.string(),
+  lastName: z.string(),
+  address1: z.string(),
+  address2: z.string().optional(),
+  city: z.string(),
+  company: z.string().optional(),
+  countryCode: z.string(),
+  stateOrProvince: z.string().optional(),
+  phone: z.string().optional(),
+  postalCode: z.string().optional(),
+  formFields: z.object({
+    checkboxes: z.array(
+      z.object({
+        fieldEntityId: stringToNumber,
+        fieldValueEntityIds: z.array(stringToNumber),
+      }),
+    ),
+    multipleChoices: z.array(
+      z.object({
+        fieldEntityId: stringToNumber,
+        fieldValueEntityId: stringToNumber,
+      }),
+    ),
+    numbers: z.array(
+      z.object({
+        fieldEntityId: stringToNumber,
+        number: stringToNumber,
+      }),
+    ),
+    dates: z.array(
+      z.object({
+        fieldEntityId: stringToNumber,
+        date: z.string(),
+      }),
+    ),
+    passwords: z.array(
+      z.object({
+        fieldEntityId: stringToNumber,
+        password: z.string(),
+      }),
+    ),
+    multilineTexts: z.array(
+      z.object({
+        fieldEntityId: stringToNumber,
+        multilineText: z.string(),
+      }),
+    ),
+    texts: z.array(
+      z.object({
+        fieldEntityId: stringToNumber,
+        text: z.string(),
+      }),
+    ),
+  }),
+});
+
+function parseAddAddressInput(
+  value: Record<string, unknown>,
+  fields: Array<Field | FieldGroup<Field>>,
+): VariablesOf<typeof AddCustomerAddressMutation>['input'] {
+  const customFields = fields
+    .flatMap((f) => (Array.isArray(f) ? f : [f]))
+    .filter(
+      (field) =>
+        field.id &&
+        ![
+          String(FieldNameToFieldId.firstName),
+          String(FieldNameToFieldId.lastName),
+          String(FieldNameToFieldId.address1),
+          String(FieldNameToFieldId.address2),
+          String(FieldNameToFieldId.city),
+          String(FieldNameToFieldId.company),
+          String(FieldNameToFieldId.countryCode),
+          String(FieldNameToFieldId.stateOrProvince),
+          String(FieldNameToFieldId.phone),
+          String(FieldNameToFieldId.postalCode),
+        ].includes(field.id),
```

Comment:
These functions/schemas are essentially the same as https://github.com/bigcommerce/catalyst/blob/soul/main/core/app/%5Blocale%5D/(default)/(auth)/register/_actions/register-customer.ts, but in this file I key off the field `id` rather than the `name` as the address form has a specific schema it adheres to.